"Manage access" modal: make sections collapsible

ajaypratap003 opened the pull request:
https://issues.redhat.com/browse/MGDSTRM-6909

Design doc: https://marvelapp.com/prototype/hc64c9a/screen/84198391

![image](https://user-images.githubusercontent.com/11775081/156814213-2938604e-084e-41cb-b7a9-ad26b25a7cf3.png)


![image](https://user-images.githubusercontent.com/11775081/156814140-eb866684-1e7e-4f7c-9d8d-75d7c2ac0f58.png)

![image](https://user-images.githubusercontent.com/11775081/156814260-0ab2533d-ccb0-4293-8e9c-16af42947da0.png)

![image](https://user-images.githubusercontent.com/11775081/156814314-6b88fb4a-6160-4d62-bf4a-748deba07899.png)

riccardo-forina commented:
Are sections staying open while interacting with the fields when adding new permissions? I remember that being a problem last time we tried adding this.

jgiardino commented on src/modules/Permissions/components/ManagePermissionsDialog/CreatePermissions.tsx:
```diff
@@ -56,6 +57,8 @@ export type CreatePermissionsProps = {
     | undefined;
   kafkaName: string | undefined;
+  isExpanded: boolean;
+  onChangeExpendedSection: (isExpanded: boolean) => void;
```
Really minor nit here, but can you fix the typo and change this to `onChangeExpandedSection` (here and throughout)?

jgiardino commented on src/modules/Permissions/dialogs/ManagePermissions/ManagePermissions.tsx:
```diff
@@ -116,6 +116,14 @@ export const ManagePermissionsModal: React.FC<
+  const [
+    isExpandedAssignPermissionsSection,
+    setIsExpandedAssignPermissionsSection,
+  ] = useState<boolean>(false);
```
I'm not sure if these values represent the default state, but when these contents first load in modal, you should see:

- **Review existing permissions** is **collapsed**
- **Assign permissions** is **expanded**

ajaypratap003 replied:
@jgiardino  yes, make sense. I'll update default state for assign permission. Thanks for the feedback.

jgiardino commented:
@ajaypratap003 - I posted some comments, but I'm not sure how to move forward on this one. The epic that contains the parent link talks about refactoring the existing UI by adding it to storybook, but this is skipping that step in the process. I think part of the reason for wanting this in storybook before adding it here was so that we could test how it behaves in the storybook preview and make sure we've addressed all issue before pushing it to stage. 

I'm going to defer to @riccardo-forina on whether he wants to approve this for merging or not. For me, not having the preview available in storybook makes me less confident in pushing changes to production.

riccardo-forina commented:
Agree on focusing the efforts on the ui-components repo

ajaypratap003 commented:
> @ajaypratap003 - I posted some comments, but I'm not sure how to move forward on this one. The epic that contains https://issues.redhat.com/browse/MGDSTRM-6909 talks about refactoring the existing UI by adding it to storybook, but this is skipping that step in the process. I think part of the reason for wanting this in storybook before adding it here was so that we could test how it behaves in the storybook preview and make sure we've addressed all issues before pushing it to stage.
> 
> I'm going to defer to @riccardo-forina on whether he wants to approve this for merging or not. For me, not having the preview available in storybook makes me less confident in pushing changes to production.

@jgiardino ExpandableSection is new feature and I don't think so there is any dependency on storybook. 
ExpandableSection can be integrate in storybook when `Manage Permissions Modal` will compete. ExpandableSection is working fine and we can merge if you are fine. 
I don't have any concern if you want to see ExpandableSection in storybook first and want to hold this change.

ajaypratap003 commented:
> Are sections staying open while interacting with the fields when adding new permissions? I remember that being a problem last time we tried adding this.

This is working fine as expected. It would be better if can try it on your machine.

riccardo-forina commented:
Let's implement this on the ui-components repo.
